yang/goldstone-nexthop.yang

```diff
@@ -0,0 +1,193 @@
+module goldstone-nexthop {
+
+  yang-version "1";
+
+  namespace "http://goldstone.net/yang/goldstone-nexthop";
+  prefix gs-nexthop;
+
+  import goldstone-interfaces { prefix gs-if; }
+
+  import ietf-inet-types {
+    prefix inet;
+  }
+
+  organization
+    "GoldStone";
+
+  description
+    "Goldstone Nexthop";
+
+  revision 2022-12-14 {
+    description
+      "Initial revision.";
+  }
+
+  typedef optional-ip-address {
+    type union {
+      type inet:ip-address;
+      type string {
+        length "0";
+      }
+    }
+  }
+
+  typedef nexthop-type {
+    type enumeration {
+      enum "ifindex" {
```

Comment:
We're using `interface-ref` and not `ifindex` to identify an interface.
I think `interface` is a better name here.